### ISSUE TITLE
DynamicProjectOptions example is incorrect

### ISSUE THREAD

**Issue by mikeloomisgg** [cpp-best-practices/project_options#86]
The dynamic project options example in the readme does not work for me. I've debugged that I must define setting defaults before including DynamicProjectOptions.cmake.

For example, in this case ccache will always be enabled even though the default should be OFF:
```
include(${_project_options_SOURCE_DIR}/src/DynamicProjectOptions.cmake)
set(ENABLE_CACHE_DEFAULT OFF)
dynamic_project_options()
```

This is resolved if you define the setting before including the macro:
```
set(ENABLE_CACHE_DEFAULT OFF)
include(${_project_options_SOURCE_DIR}/src/DynamicProjectOptions.cmake)
dynamic_project_options()
```

I don't know if this behavior can be changed, so that the setting will always be correctly overridden by the dynamic_project_options macro. But I do think the example shown in the readme should be fixed.

**Comment by mikeloomisgg** [comment]
To be clear, if you run the example as written in the readme it *does* function. PCH header compilation behavior will correctly follow the predefined defaults.

It is the setting override behavior that doesn't not work as described.

**Comment by aminya** [comment]
This was fixed in #80. But it is not released yet. I am planning to add another PR that forwards the options as function arguments. This is much better than dancing around with global CMake variables.

**Comment by mikeloomisgg** [comment]
I see, I had seen this on 0.15 and 0.17, glad to know it's already fixed.
